contracts/BuyBackBurner.sol

```diff
@@ -117,10 +118,23 @@ contract BuyBackBurner {
         require(pool != address(0), "Pool does not exist");
 
         // Get current pool reserves
-        (uint160 sqrtPriceX96, , , , , , ) = IUniswapV3(pool).slot0();
+        (uint160 sqrtPriceX96, , uint16 observationCardinality, , , , ) = IUniswapV3(pool).slot0();
+        // Check observation cardinality
+        if (observationCardinality < 2) {
+            // Increase observation cardinality to get more accurate twap
+            IUniswapV3(pool).increaseObservationCardinalityNext(60);
+            return;
+        }
```

Comment:
The default pool observation cardinality is 1, and it's not enough to do twap-s correctly. While it's still 1 the instant price is fine for consideration.